.github/workflows/CI.yml

```diff
@@ -0,0 +1,92 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - 'main'
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@2c7b3805fd2a0fd8c1884dcaebf91fc102a13ecd #v3.0.1
+
+  pypi-build:
+    name: Build package for PyPI
+    if: github.repository == 'access-nri/model-config-tests'
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+          cache: 'pip'
+
+      - name: Install build dependencies
+        run: python3 -m pip install --upgrade build
+
+      - name: Build distributions
+        run: python3 -m build
+
+      - name: Upload distribution artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: release
+          path: dist
+
+  install_from_dist:
+    needs: [ 'pypi-build' ]
+    name: Test install from distribution
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download distribution
+        uses: actions/download-artifact@v4
+        with:
+          name: release
+          path: dist
+
+      - name: Setup Python 3.11
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install model_config_tests
+        run: python3 -m pip install ./dist/model_config_tests-*.tar.gz
+
+      - name: Run model-config-tests command
+        run: model-config-tests --collect-only -m "checksum"
```

Comment:
If we're running this on GitHub is`checksum` the tests we want?

Reply:
I am just testing for dependency runtime errors hence the --collect-only but it doesn't run the tests. Though I could just change it to `model-config-tests --collect-only` and it prints all the tests in the src code.